src/jit/codegencommon.cpp

```diff
@@ -4426,32 +4447,32 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
 
         if (doingFloat)
         {
-#if defined(_TARGET_ARM_) || defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
-            insCopy = ins_Copy(TYP_FLOAT);
+#if defined(FEATURE_HFA) || defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
+            insCopy = ins_Copy(TYP_DOUBLE);
             // Compute xtraReg here when we have a float argument
```

Comment:
I am not knowledgeable about this, but it seems surprising that this could be changed from TYP_DOUBLE to TYP_FLOAT for _TARGET_ARM_